Mathlib/Probability/Distributions/Uniform.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2024 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker, Kexing Ying
+-/
+import Mathlib.Probability.Notation
+import Mathlib.Probability.Cdf
+import Mathlib.Probability.Density
+
+/-! # Uniform distributions
+
+Defines the uniform distribution for any set with finite measure.
+
+## Main definitions
+* `uniformMeasure s` : The uniform measure on `s` is the
+  the measure restricted to `s`, normalized.
+* `IsUniform X s ℙ μ` : A random variable `X` has uniform distribution on `s` under `ℙ` if the
+  push-forward measure agrees with the rescaled restricted measure `μ`.
+-/
+
+open scoped Classical MeasureTheory NNReal ENNReal
+
+open TopologicalSpace MeasureTheory.Measure
+
+noncomputable section
+
+namespace MeasureTheory
+
+variable {E : Type*} [MeasurableSpace E] {m : Measure E} {μ : Measure E}
+
+/-- A measure is a uniform measure for a set `s` if it is the rescaled restriction of the measure to
+this set.  -/
+def uniformMeasure (s : Set E) (μ : Measure E) : Measure E := (μ s)⁻¹ • μ.restrict s
+
+namespace UniformMeasure
+
+theorem absolutelyContinuous {s : Set E} :
+    uniformMeasure s μ ≪ μ := by
+  intro t ht
+  unfold uniformMeasure
+  rw [smul_apply, smul_eq_mul]
+  apply mul_eq_zero.mpr
+  refine Or.inr (le_antisymm ?_ (zero_le _))
+  exact ht ▸ restrict_apply_le s t
+
+theorem uniformMeasure_apply {s : Set E} {A : Set E}
+    (hA : MeasurableSet A) : uniformMeasure s μ A = μ (s ∩ A) / μ s := by
+  unfold uniformMeasure
+  rw [smul_apply, restrict_apply hA, ENNReal.div_eq_inv_mul, smul_eq_mul, Set.inter_comm]
+
+theorem isProbabilityMeasure {s : Set E} (hns : μ s ≠ 0) (hnt : μ s ≠ ∞) :
+    IsProbabilityMeasure (uniformMeasure s μ) :=
+  ⟨by
+    unfold uniformMeasure
+    simp only [smul_toOuterMeasure, OuterMeasure.coe_smul, Pi.smul_apply, MeasurableSet.univ,
+      restrict_apply, Set.univ_inter, smul_eq_mul]
+    exact ENNReal.inv_mul_cancel hns hnt⟩
+
+theorem toMeasurable_eq {s : Set E} :
+    uniformMeasure (toMeasurable μ s) μ = uniformMeasure s μ:= by
+  unfold uniformMeasure
+  by_cases hnt : μ s = ∞
+  · simp [hnt]
+  · simp [restrict_toMeasurable hnt]
+
+end UniformMeasure
+
+namespace pdf
+
+variable {Ω : Type*}
+
+variable {_ : MeasurableSpace Ω} {ℙ : Measure Ω}
+
+/-- A random variable `X` has uniform distribution on `s` if its push-forward measure is
+`(μ s)⁻¹ • μ.restrict s`. -/
+def IsUniform (X : Ω → E) (s : Set E) (ℙ : Measure Ω) (μ : Measure E) :=
```

Comment:
Why did you remove `by volume_tac`? The role of `by volume_tac` is to automatically fill the measure with `volume`, in case the space is actually a measure space and a volume is available (which is the main case we care about). It does not prevent you from using the def with other measures.

Reply:
Because I misunderstood by volume_tac, my bad. I'll put it back!